### PR TITLE
nRF53 network CPU on-off management

### PR DESCRIPTION
**Pull request by jciupis**
This PR addresses the problem of safely sharing nRF53 network CPU as a shared resource used by multiple users on the application CPU. The current solution, where every user controls the network CPU directly, scales poorly and leads to hidden dependencies between those users.

### CODE REVIEW

**Comment by jciupis** [comment]
FYI: @ankuns, @vidarbe

**Review by anangl — DISMISSED** [review]

**Review by rerickson1 — DISMISSED** [review]

**Review by nika-nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Comment by jciupis** [comment]
Rebased, corrected the API documentation and used the API for two more boards that relied on direct register access, which I missed earlier.

**Review by anangl — DISMISSED** [review]

**Comment by jciupis** [comment]
To correctly handle use cases where the application doesn't use `CONFIG_ONOFF`, a separate Kconfig entry that selects it was added.

**Review by kl-cruz — APPROVED** [review]

File: soc/nordic/nrf53/Kconfig [hunk 1]
```diff
@@ -138,6 +139,13 @@ config SOC_NRF_GPIO_FORWARDER_FOR_NRF5340
 	help
 	  hidden option for including the nRF GPIO pin forwarding
 
+config SOC_NRF53_CPUNET_MGMT
+	bool
+	default y
```
Comment (anangl) [inline soc/nordic/nrf53/Kconfig:144]: nit: there is no need to both make this option `default y` and also `select` it above

File: soc/nordic/nrf53/nrf53_cpunet_mgmt.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+#include <nrf53_cpunet_mgmt.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/notify.h>
+#include <zephyr/sys/onoff.h>
+
+#include <soc.h>
```
Comment (anangl) [inline soc/nordic/nrf53/nrf53_cpunet_mgmt.c:19]: Is it actually needed to include this?

**Review by anangl — DISMISSED** [review]

File: soc/nordic/nrf53/nrf53_cpunet_mgmt.c (hunk 2 above)
Comment (jciupis) [reply soc/nordic/nrf53/nrf53_cpunet_mgmt.c]: No, good catch. I removed it

File: soc/nordic/nrf53/Kconfig (hunk 1 above)
Comment (jciupis) [reply soc/nordic/nrf53/Kconfig]: Removed `default y`

**Review by anangl — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Pull request merged**